services/shhext/api.go

```diff
@@ -2,30 +2,84 @@ package shhext
 
 import (
 	"context"
+	"crypto/ecdsa"
+	"encoding/binary"
+	"errors"
 	"fmt"
+	"time"
 
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
 	"github.com/ethereum/go-ethereum/p2p/discover"
 	whisper "github.com/ethereum/go-ethereum/whisper/whisperv6"
 )
 
-// NewPublicAPI returns instance of the public API.
-func NewPublicAPI(w *whisper.Whisper, tracker *tracker) *PublicAPI {
-	return &PublicAPI{
-		w:       w,
-		tracker: tracker,
+const (
+	// defaultWorkTime is a work time reported in messages sent to MailServer nodes.
+	defaultWorkTime = 5
+)
+
+var (
+	// ErrInvalidMailServerPeer is returned when it fails to parse enode from params.
+	ErrInvalidMailServerPeer = errors.New("invalid mailServerPeer value")
+	// ErrInvalidSymKeyID is returned when it fails to get a symmetric key.
+	ErrInvalidSymKeyID = errors.New("invalid symKeyID value")
+)
+
+// -----
+// PAYLOADS
+// -----
+
+// MessagesRequest is a payload send to a MailServer to get messages.
+type MessagesRequest struct {
+	// MailServerPeer is MailServer's enode address.
+	MailServerPeer string `json:"mailServerPeer"`
+
+	// From is a lower bound of time range (optional).
+	// Default is 24 hours back from now.
+	From uint32 `json:"from"`
+
+	// To is a upper bound of time range (optional).
+	// Default is now.
+	To uint32 `json:"to"`
+
+	// Topic is a regular Whisper topic.
+	Topic whisper.TopicType `json:"topic"`
+
+	// SymKeyID is an ID of a symmetric key to authenticate to MailServer.
+	// It's derived from MailServer password.
+	SymKeyID string `json:"symKeyID"`
+}
+
+func (r *MessagesRequest) setDefaults() {
+	// set From and To defaults
+	if r.From == 0 && r.To == 0 {
+		r.From = uint32(time.Now().UTC().Add(-24 * time.Hour).Unix())
+		r.To = uint32(time.Now().UTC().Unix())
 	}
 }
 
+// -----
```

Comment:
Not sure, it's better to put this kind of comments instead of splitting into multiple files :P But as it was already changed twice, let's leave it as it is.

Reply:
As long as the file is small enough it's IMHO okay. But later we've got to keep an eye on it, yes.

Had seen these block building comments in larger projects and learned them as being quite helpful when later have to orient in a source code. So adopted them.